### PR TITLE
Add New Search Space MoSegNAS & New Benchmark Test Suite CitySeg/MOP

### PR DESCRIPTION
**Pull request by AlpAcA0072**
Add a new search space MoSegNAS and a new Benchmark Test Suite CitySeg/MOP to EvoXBench.
Code changes are minimized and formatter tools `black` was used to format the code.

### CODE REVIEW

File: rpc/matlab/benchmark_citysegmop.m [hunk 1]
```diff
@@ -0,0 +1,35 @@
+addpath(genpath(cd));
+clear;
+clc;
+
+problem_list = {@CITYMOP1, @CITYMOP2, @CITYMOP3, @CITYMOP4, @CITYMOP5, @CITYMOP6, @CITYMOP7, @CITYMOP8, @CITYMOP9, @CITYMOP10, @CITYMOP11, @CITYMOP12, @CITYMOP13, @CITYMOP14, @CITYMOP15};
```
Comment (Zhenyu2Liang) [inline rpc/matlab/benchmark_citysegmop.m]: The problem names are best modified as **CITYSEGMOP**.

File: rpc/matlab/benchmark_citysegmop.m [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+algorithms = {@NSGAII, @NSGAIII, @MOEAD, @RVEAa, @SMSEMOA};
+algorithm_names = {'nsga2', 'nsga3', 'moead', 'RVEAa', 'smsemoa'};
+pop_size = {100, 105, 105, 120, 126, 100, 105, 105, 120, 126, 105, 126, 132, 132, 217};
+num_runs = 31;  % the number of runs
+
+for algidx = 1 : 6
```
Comment (Zhenyu2Liang) [inline rpc/matlab/benchmark_citysegmop.m]: There are only 5 algorithms in the script.

File: tutorial/benchmark_citysegmop.py [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+from pymoo.algorithms.moo.rvea import RVEA
+from pymoo.algorithms.moo.sms import SMSEMOA
+from pymoo.algorithms.moo.spea2 import SPEA2
+
+from pymoo.operators.crossover.sbx import SBX
+from pymoo.operators.mutation.pm import PM
+from pymoo.operators.sampling.rnd import IntegerRandomSampling
+from pymoo.operators.repair.rounding import RoundingRepair
+
+from evoxbench.test_suites.citymop import citymop
```
Comment (Zhenyu2Liang) [inline tutorial/benchmark_citysegmop.py]: Wrong import.

File: tutorial/benchmark_citysegmop.py [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+
+class C10MOPProblem(Problem):
```
Comment (Zhenyu2Liang) [inline tutorial/benchmark_citysegmop.py]: Class name change to CitySegProblem.

Comment (AlpAcA0072) [reply tutorial/benchmark_citysegmop.py]: Thanks for correction！

**Pull request merged**